src/net_message.h

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_NET_MESSAGE_H
+#define BITCOIN_NET_MESSAGE_H
+
+#include <hash.h>
+#include <protocol.h>
+#include <streams.h>
+#include <uint256.h>
+
+/** Maximum length of incoming protocol messages (no message over 4 MB is currently acceptable). */
+static const unsigned int MAX_PROTOCOL_MESSAGE_LENGTH = 4 * 1000 * 1000;
+
+// base class for format agnostic network messages
+class NetMessageBase
+{
+public:
+    CDataStream vRecv; // received message data
+    int64_t nTime;     // time (in microseconds) of message receipt.
```

Comment:
According to https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md these should be prefixed `m_`.

Reply:
I kept it because it is moved code and to simplify the review.

Reply:
This is moved code right?